syfertext/tokenizer.py

```diff
@@ -175,18 +217,411 @@ def __call__(self, text: Union[String, str]):
+
+    def _attach_tokens(
+        self,
+        doc: Doc,
+        substring: str,
+        start_pos: int,
+        space_after: bool,
+        affixes: DefaultDict,
+        exception_tokens: List[TokenMeta],
+    ) -> Doc:
+        """Attach all the `TokenMeta` objects which are the result of splitting affixes
+        in Doc object's container. Returns Doc object.
+       
+        Args:
+            doc: Original Document
+            substring: The substring remaining after splitting all the affixes.
+            start_pos: The pointer to location of start of substring in text.
+            space_after: If there is a space after the original substring before splitting any affixes 
+                in the text.
+            affixes: Dict holding TokenMeta lists of each affix types(prefix, suffix, infix) 
+                formed as the result of splitting affixes.
+            exception_tokens: The list of TokenMeta object of exception tokens.
+
+        Returns:
+            doc: Document with all the TokenMeta objects of every token after splitting 
+                affixes and exceptions.
+        """
+
+        # Append the prefix TokenMeta list in doc
+        doc.container.extend(affixes["prefix"])
+
+        # Append the exceptions TokenMeta list in doc
+        doc.container.extend(exception_tokens)
+
+        # If subtring is remaining after splitting all the affixes.
+        if substring:
+            # Create the TokenMeta object
+            end_pos = start_pos + len(substring) - 1
+            token_meta = TokenMeta(
+                hash_key=self.vocab.store[(substring)],
+                start_pos=start_pos,
+                end_pos=end_pos,
+                space_after=False,  # for the last token space_after will be updated explicitly according to the original substring.
+                is_space=False,
+            )
+
+            # Append the token to the document
+            doc.container.append(token_meta)
+
+        # Append the infixes TokenMeta list in doc
+        doc.container.extend(affixes["infix"])
+
+        # Append the suffixes TokenMeta list in doc
+        doc.container.extend(reversed(affixes["suffix"]))
+
+        # Get the last token and update it's space_after attr according to original substring's TokenMeta data
+        doc.container[-1].space_after = space_after
+
+        return doc
+
+    def _get_prefix_token_meta(self, substring: str, pos: int) -> Tuple[TokenMeta, str, int]:
+        """Makes TokenMeta data for substring which are prefixes.
+
+        Args:
+            substring: The substring to tokenize.
+            pos: The pointer to the start position of substring in the text.
+
+        Returns:
+            token_meta: The TokenMeta object with TokenMeta data of prefix.
+            substring: The updated substring after removing prefix.
+            pos: The pointer to the start position of new substring in the text.
+        """
+        pre_len = self.find_prefix(substring)
```

Comment:
please add an empty line between the docstring and this line.

Also, comment this line please. 

And add an empty line after it as well.